### PR TITLE
[Tests] migrate to Github Actions

### PR DESCRIPTION
**Pull request by ljharb**
Unblocks #2084.

### CODE REVIEW

File: .github/workflows/node-pretest.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ljharb/actions/node/install@main
+        name: 'nvm install lts/* && npm install'
+        with:
+          node-version: 'lts/*'
+      - run: npm run build
```
Comment (sharmilajesupaul) [inline .github/workflows/node-pretest.yml:37]: nit. end on a newline

File: .github/workflows/node-pretest.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  # posttest:
+  #   runs-on: ubuntu-latest
+
+  #   steps:
+  #     - uses: actions/checkout@v2
+  #     - uses: ljharb/actions/node/install@main
+  #       name: 'nvm install lts/* && npm install'
+  #       with:
+  #         node-version: 'lts/*'
+  #     - run: npm run posttest
```
Comment (sharmilajesupaul) [inline .github/workflows/node-pretest.yml:26]: Did you mean to leave this block in?

File: .travis.yml [hunk 3]
```diff
@@ -1,43 +1,13 @@
...
-      env: BUILD=true TEST=false
-    - node_js: "lts/*"
-      env: COVERAGE=true TEST=false
-    - node_js: "lts/*"
-      env: KARMA=true TEST=false
-  allow_failures:
-    - node_js: "10"
-      env: REACT=16
+    - DISPLAY=:99.0
+sudo: false
```
Comment (sharmilajesupaul) [inline .travis.yml:13]: same here, end on newline

**Review by sharmilajesupaul — APPROVED** [review]

**Comment by sharmilajesupaul** [comment]
@lencioni / @madicap could you take a look here? coveralls seems to be blocking since it's being removed from travis.yml in favor of codecov. So we need to remove the check from this repo's branch protection.

File: .github/workflows/node-pretest.yml (hunk 2 above)
Comment (ljharb) [reply .github/workflows/node-pretest.yml:26]: yes; i plan to uncomment it in the future.

**Comment by ljharb** [comment]
(If the branch protections don't get updated in the next day or two, I'll re-add coveralls (alongside codecov) just to make the protections happy, so that we can unblock #2084.

**Comment by ljharb** [comment]
Looks like coveralls isn't even working anymore - the engineer who connected it probably left airbnb, had their repo admin permissions removed, and thus the coveralls app lost the ability to update PR statuses - so this will have to wait until the branch protections can be fixed.

**Pull request merged**